Source/Diagnostics/Diagnostics.cpp

```diff
@@ -452,6 +452,45 @@ Diagnostics::InitData ()
     }
 }
 
+void
+Diagnostics::InitializeParticleBuffer ()
```

Comment:
This function has the combined logic from the old `BTDiagnostic::InitializeParticleBuffer` and `BoundaryScrapingDiagnostics::InitializeParticleBuffer`